helm/cluster-api-provider-azure/templates/_helpers.tpl

```diff
@@ -36,3 +36,22 @@ Provider labels
 {{- define "labels.provider" -}}
 cluster.x-k8s.io/provider: infrastructure-azure
 {{- end -}}
+
+{{- define "capz.crdInstall" -}}
+{{- printf "%s-%s" ( include "name" . ) "crd-install" | replace "+" "_" | trimSuffix "-" -}}
+{{- end -}}
+
+{{- define "capz.CRDInstallAnnotations" -}}
+"helm.sh/hook": "pre-install,pre-upgrade"
+"helm.sh/hook-delete-policy": "before-hook-creation,hook-succeeded,hook-failed"
```

Comment:
should we keep the resources when it fails? I mean removing `hook-failed` from here.